### PR TITLE
Non Interactive Proof of Replication

### PR DESCRIPTION
**Pull request by NemanjaLu92**

### CODE REVIEW

File: shared/src/sector/seal.rs [hunk 1]
```diff
@@ -33,6 +33,18 @@ pub struct SealVerifyInfo {
...
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct NISealVerifyInfo {
+    pub registered_proof: RegisteredSealProof,
+    pub sector_id: SectorID,
+    pub randomness: SealRandomness,
+    #[serde(with = "strict_bytes")]
+    pub proof: Vec<u8>,
+    pub sealed_cid: Cid,   // Commr
+    pub unsealed_cid: Cid, // Commd
+}
```
Comment (anorth) [inline shared/src/sector/seal.rs:46]: Can we do this without introducing a new NISealVerifyInfo type, and thus without a new syscall at all? The proof type carries the information that it is NI, and then determines that certain fields will be ignored.
- deal_ids: already ignored and always empty
- interactive_randomness: let's just specify it shall be zero (and ignored) for NI proofs

Comment (NemanjaLu92) [reply shared/src/sector/seal.rs:46]: Yeah, that was my initial thinking too. Will change that.

**Comment by anorth** [comment]
This is probably ready for review. We'll need it to land so the actors change can depend on it.

@Stebalien the associated FIP is accepted. Can we land this direct to master?

**Comment by Stebalien** [comment]
It looks like this doesn't build with the "proofs" feature enabled.

**Comment by rvagg** [comment]
just needs the additional `TryFrom` matches added I believe

**Comment by NemanjaLu92** [comment]
> just needs the additional `TryFrom` matches added I believe

True, but I need to change dependancy to the `filecoin-proofs-api` which doesn't include NI PoRep Registered proof in latest version (17.0). How should we proceed? Create a new release in `filecoin-proofs-api` and update dependency in ref-fvm?

**Comment by rjan90** [comment]
> True, but I need to change dependancy to the filecoin-proofs-api which doesn't include NI PoRep Registered proof in latest version (17.0). How should we proceed? Create a new release in filecoin-proofs-api and update dependency in ref-fvm?

A new Filecoin-proofs-api release is expected to be out today (Thursday May 16th), which includes the NI-PoRep registered proofs.

**Comment by Stebalien** [comment]
It looks like `storage-proofs-update` is either missing a minimum version specification for something, or it isn't enabling some feature flag... @cryptonemo?

**Comment by rjan90** [comment]
Can we close this, since its superseded by https://github.com/filecoin-project/ref-fvm/pull/2010?

**Comment by Stebalien** [comment]
Yes.

**Pull request closed without merging**